flow/util/misc.py

```diff
@@ -251,3 +251,50 @@ def _cached_partial(func, *args, maxsize=None, **kwargs):
 
     """
     return lru_cache(maxsize=maxsize)(partial(func, *args, **kwargs))
+
+
+class _bidict(dict):
```

Comment:
@bdice Also do note that `update` will not call `__setitem__`, `pop` will not call `__delitem__`, `setdefault` will not call `__setitem__` and so on. The CPython `dict` class is optimized to not use these paradigms that exist in `abc.collections`. We should probably either disable these methods `return NotImplementedError` or implement all methods that involve setting or deleting. I forgot this when reviewing last night.

Reply:
@b-butler That's a really helpful observation, I didn't realize that. I re-implemented the class as a `MutableMapping` with an internal dictionary, so it gets all of those methods "for free." Also, I added tests for all of the methods expected from a `MutableMapping`. Fixed in 0789f2a.